src/components/molecules/EventDisplay/EventDisplay.tsx

```diff
@@ -16,21 +17,26 @@ interface EventDisplayProps {
 }
 
 export const EventDisplay: React.FC<EventDisplayProps> = ({ event, venue }) => {
-  const room = venue?.rooms?.find((room) => room.title === event?.room);
+  const eventRoomTitle = event.room;
+
+  const room = useMemo(
+    () => venue?.rooms?.find((room) => room.title === eventRoomTitle),
+    [eventRoomTitle]
+  );
+
   const buttonText = `${event.room ?? "Enter"} ${venue && `- ${venue.name}`}`;
 
   const isLiveEvent =
     event.start_utc_seconds < getCurrentTimeInUTCSeconds() &&
     event.start_utc_seconds + event.duration_minutes * 60 >
       getCurrentTimeInUTCSeconds();
 
+  const containerClasses = classNames("schedule-event-container", {
+    "schedule-event-container_live": isLiveEvent,
+  });
+
   return (
-    <div
-      key={event.name + Math.random().toString()}
-      className={`schedule-event-container ${
-        isLiveEvent && "schedule-event-container_live"
-      }`}
-    >
+    <div key={event.name} className={containerClasses}>
```

Comment:
This shouldn't need the `key` prop at all, as it's not within a loop.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1129/commits/53f57c40e24983470b7552d4ebed6f1715ff74ba